### PR TITLE
Simplify investment manager

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: src/InvestmentManager.sol [hunk 1]
```diff
@@ -318,118 +280,99 @@ contract InvestmentManager is Auth {
         public
         onlyGateway
     {
-        address currency = poolManager.currencyIdToAddress(currencyId);
-        address liquidityPool = poolManager.getLiquidityPool(poolId, trancheId, currency);
-        require(liquidityPool != address(0), "InvestmentManager/tranche-does-not-exist");
-
+        address liquidityPool = poolManager.getLiquidityPool(poolId, trancheId, currencyId);
```
Comment (ilinzweilin) [inline src/InvestmentManager.sol:283]: 👍

File: src/InvestmentManager.sol [hunk 2]
```diff
@@ -773,76 +612,36 @@ contract InvestmentManager is Auth {
...
-        uint256 newMaxWithdraw = _toPriceDecimals(currentMaxWithdraw + currencyPayout, currencyDecimals);
-        if (newMaxWithdraw == 0) redeemPrice = 0;
-        else redeemPrice = newMaxWithdraw.mulDiv(10 ** PRICE_DECIMALS, newMaxRedeem, MathLib.Rounding.Down);
-    }
-
     function _calculatePrice(address liquidityPool, uint128 currencyAmount, uint128 trancheTokenAmount)
         public
         view
-        returns (uint128 price)
+        returns (uint256 price)
```
Comment (ilinzweilin) [inline src/InvestmentManager.sol:618]: 👍

**Review by ilinzweilin — APPROVED** [review]

**Review by AStox — DISMISSED** [review]

File: src/InvestmentManager.sol [hunk 3]
```diff
@@ -181,63 +171,67 @@ contract InvestmentManager is Auth {
...
+            "InvestmentManager/currency-not-allowed"
+        );
 
-        LPValues storage lpValues = orderbook[liquidityPool][user];
-        lpValues.remainingRedeemOrder = lpValues.remainingRedeemOrder + _trancheTokenAmount;
-        lpValues.exists = true;
+        _processIncreaseRedeemRequest(liquidityPool, _trancheTokenAmount, user);
+    }
+
+    function _processIncreaseRedeemRequest(address liquidityPool, uint128 trancheTokenAmount, address user) internal {
```
[the remark this answers is not in the document]
Comment (hieronx) [reply src/InvestmentManager.sol:206]: > Or are we just assuming if you're redeeming, then you've already deposited and set exists = true?

Yes exactly this :) You should not be able to do `requestRedeem` before you ever did `requestDeposit`

[the remark this answers is not in the document]
Comment (hieronx) [reply src/InvestmentManager.sol:206]: Actually I just realized that might not always hold. In theory you could invest on another domain, transfer the tranche tokens manually, and then redeem on this chain.

**Review by AStox — APPROVED** [review]

**Pull request merged**